### PR TITLE
Reduce e2e tests flakiness

### PR DESCRIPTION
**Pull request by sebastinez**
This PR removes all of the sleep instructions from the seed fixture creation, and is able to rely on checking node events and node routing tables.

This allows the tests to await the necessary events and routing table entries instead of sleeping and hoping for them.

### CODE REVIEW

File: tests/support/peerManager.ts [hunk 1]
```diff
@@ -89,6 +92,14 @@ export class RadiclePeer {
     this.#outputLog = props.logFile;
   }
 
+  public waitForEvent(searchEvent: NodeEvent, timeoutInMs: number) {
+    const start = new Date().getTime();
+
+    while (new Date().getTime() - start > timeoutInMs) {
+      this.#eventRecords.find(event => lodash.isEqual(searchEvent, event));
```
Comment (rudolfs) [inline tests/support/peerManager.ts:99]: I'd add a 100ms sleep here, otherwise this will load the CPU 100%.

File: tests/support/peerManager.ts [hunk 2]
```diff
@@ -164,9 +168,69 @@ export class RadiclePeer {
...
+              } catch (e) {
+                console.log("Error parsing event", data);
+              }
+            }
+          });
+      },
+    );
+  }
+
+  public async routesTo(rid: string, ...nodes: string[]) {
```
Comment (rudolfs) [inline tests/support/peerManager.ts:194]: Could we call this function `addRoute`?

**Review by rudolfs — COMMENTED** [review]
Nice work!

Noticed that if we don't have the binaries downloaded and run the `e2e` tests, we don't print instructions to the developer on how to download the binaries. Would be nice if we could add something like (this could also be a follow-up):

```
npm run test:e2e -- --project chromium

> test:e2e
> TZ='UTC' playwright test --project chromium

Error: rad version rad 0.8.0 (fc951b8) does not satisfy 29507b5
    at assertRadInstalled (file:///Users/rudolfs/work/radicle-interface/tests/support/globalSetup.ts:47:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
...

To download the required test binaries, run:
  👉 ./scripts/install-binaries

```

File: tests/support/peerManager.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
 /* eslint-disable @typescript-eslint/naming-convention */
 import type { ExecaChildProcess, Options } from "execa";
+import type { NodeEvent, RoutingEntry } from "./cliTypes.js";
```
[the remark this answers is not in the document]
Comment (sebastinez) [reply tests/support/peerManager.ts:3]: Yeah sure, didn't want to overload the `peerManager` file.

File: tests/support/peerManager.ts (hunk 1 above)
Comment (sebastinez) [reply tests/support/peerManager.ts:99]: Sure

File: tests/support/peerManager.ts (hunk 2 above)
Comment (sebastinez) [reply tests/support/peerManager.ts:194]: I have no issue with renaming the function, but the goal is to wait for the routing table to have a specific entry.
So maybe we could rename it to `waitForRoutes` ? So it's similar to `waitForEvent`wdyt?

Comment (rudolfs) [reply tests/support/peerManager.ts:194]: Yeah, `waitForRoutes` makes more sense.

File: tests/support/peerManager.ts (hunk 1 above)
Comment (sebastinez) [reply tests/support/peerManager.ts:137]: e4c2861

File: tests/support/peerManager.ts (hunk 2 above)
Comment (sebastinez) [reply tests/support/peerManager.ts]: e4c2861

**Comment by sebastinez** [comment]
> Nice work!
> 
> Noticed that if we don't have the binaries downloaded and run the `e2e` tests, we don't print instructions to the developer on how to download the binaries. Would be nice if we could add something like (this could also be a follow-up):
> 
> ```
> npm run test:e2e -- --project chromium
> 
> > test:e2e
> > TZ='UTC' playwright test --project chromium
> 
> Error: rad version rad 0.8.0 (fc951b8) does not satisfy 29507b5
>     at assertRadInstalled (file:///Users/rudolfs/work/radicle-interface/tests/support/globalSetup.ts:47:11)
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
> ...
> 
> To download the required test binaries, run:
>   👉 ./scripts/install-binaries
> ```

Good idea, addressed it with e4c2861

**Review by rudolfs — APPROVED** [review]

**Pull request merged**